lib/plugins/bower-manifest.js

```diff
@@ -0,0 +1,61 @@
+import processJSON from './helper/process-json';
+import { isSemver } from './helper/version';
+import JavaScriptPlugin from './javascript';
+import { jsonRegExKeyValue, jsonRegExValue } from './helper/regex-builder';
+import insertLink from '../insert-link';
+import liveResolverQuery from '../resolver/live-resolver-query.js';
+import gitUrl from '../resolver/git-url.js';
+import githubShorthand from '../resolver/github-shorthand.js';
+
+function linkDependency(type, blob, key, value) {
+  const isValidSemver = isSemver(value);
+  const regex = jsonRegExKeyValue(key, value);
+
+  insertLink(blob.el, regex, {
+    pluginName: 'BowerManifest',
+    target: isValidSemver ? '$1' : '$2',
+    type: isValidSemver ? 'liveResolverQuery' : 'git',
+  });
+}
+
+function linkFile(blob, key, value) {
+  const regex = jsonRegExValue(key, value);
+
+  insertLink(blob.el, regex, {
+    pluginName: 'BowerManifest',
+    type: 'file',
+    path: blob.path,
+    target: '$1',
+  });
+}
+
+export default class BowerManifest {
+
+  static resolve({ target, path, type }) {
+    if (type === 'file') {
+      return JavaScriptPlugin.resolve({ target, path });
+    }
+
+    if (type === 'liveResolverQuery') {
+      return liveResolverQuery({ type: 'bower', target });
+    }
+
+    return [
+      gitUrl({ target }),
+      githubShorthand({ target }),
+    ];
+  }
```

Comment:
I'm talking about the if statements in the resolve function.  In my current implementation, the `type` attributes define which resolver is used. Without knowing how the click-handler works, I would assume that every `insertLink` can have his own click handler. What do you think? Maybe I'm over-thinking it.